test/TestProjects/SingleTopLevelClientWithOperations-LowLevel/Generated/TopLevelClientWithOperationClient.cs

```diff
@@ -0,0 +1,134 @@
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <param name="endpoint"> server parameter. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="credential"/> is null. </exception>
+        public TopLevelClientWithOperationClient(AzureKeyCredential credential, Uri endpoint = null, TopLevelClientWithOperationClientOptions options = null)
+        {
+            if (credential == null)
+            {
+                throw new ArgumentNullException(nameof(credential));
+            }
+            endpoint ??= new Uri("http://localhost:3000");
+            options ??= new TopLevelClientWithOperationClientOptions();
+
+            _clientDiagnostics = new ClientDiagnostics(options);
+            _keyCredential = credential;
+            _pipeline = HttpPipelineBuilder.Build(options, new HttpPipelinePolicy[] { new LowLevelCallbackPolicy() }, new HttpPipelinePolicy[] { new AzureKeyCredentialPolicy(_keyCredential, AuthorizationHeader) }, new ResponseClassifier());
+            _endpoint = endpoint;
+        }
+
+        /// <param name="context"> The request context. </param>
+#pragma warning disable AZC0002
+        public virtual async Task<Response> OperationAsync(RequestContext context = null)
+#pragma warning restore AZC0002
+        {
+            using var scope = _clientDiagnostics.CreateScope("TopLevelClientWithOperationClient.Operation");
+            scope.Start();
+            try
+            {
+                using HttpMessage message = CreateOperationRequest();
+                return await _pipeline.ProcessMessageAsync(message, _clientDiagnostics, context).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <param name="context"> The request context. </param>
+#pragma warning disable AZC0002
+        public virtual Response Operation(RequestContext context = null)
+#pragma warning restore AZC0002
+        {
+            using var scope = _clientDiagnostics.CreateScope("TopLevelClientWithOperationClient.Operation");
+            scope.Start();
+            try
+            {
+                using HttpMessage message = CreateOperationRequest();
+                return _pipeline.ProcessMessage(message, _clientDiagnostics, context);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        private volatile Client1Client _cachedClient1Client;
+        private volatile Client2Client _cachedClient2Client;
+
+        /// <summary> Initializes a new instance of Client1Client. </summary>
+        public virtual Client1Client GetClient1ClientClient()
```

Comment:
Can we add some unit tests to verify generated methods for single top-level client?

Reply:
Done